packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder;
+
+import 'package:flutter/foundation.dart';
+import 'package:meta/meta.dart';
+
+import 'basic_types.dart';
+import 'text_painter.dart';
+import 'text_style.dart';
+
+/// Mutable wrapper of an integer that can be passed by reference to track a
+/// value across a recursive stack.
+class TrackingInt {
```

Comment:
I am preliminarily calling this TrackingInt, but I am not sure what the precedent for naming/usage of a wrapper class that allows me to pass primitives with the behavior of passing by references.

Reply:
cc @cbracken

Reply:
I think we would make a specialized class for the specific case. See for instance HitTestResults.